src/Database/Attach/Resizer.php

```diff
@@ -114,11 +114,14 @@ public static function open($file)
      * Manipulate an image resource in order to keep transparency for PNG and GIF files.
      * @param $img
      */
-    protected static function retainImageTransparency($img)
+    protected static function retainImageTransparency($img, $extension)
```

Comment:
@mjauvin I'd prefer the MIME type being used to determine which image calls to use here, and in all the other places that extension is being checked and used, as the MIME type is (a little) harder to spoof.

Reply:
done